lib/event_hub/adapters/aws/message.rb

```diff
@@ -0,0 +1,53 @@
+class EventHub::Adapters::Aws::Message < EventHub::Message
+  def initialize(adapter, message)
+    @adapter = adapter
+    @message = message
+  end
+
+  def attributes
+    @headers ||= parsed_body['MessageAttributes'].transform_values { |v| v['Value'] }
+  end
+
+  def body
+    parsed_body['Message']
+  end
+
+  def event
+    attributes['event']
+  end
+
+  def version
+    attributes['version']
+  end
+
+  def ack
+    # Delete the message from the queue.
+    @adapter.delete_message(@message.receipt_handle)
+  end
+
+  def reject
+    if @adapter.config[:dead_queue_url]
+      # TODO: publish the message to the dead_queue
```

Comment:
currently as I can see AWS automatically sends the message into the dead queue if it was returned into the main queue the specified amount of times. I didn't find a way to do that by the AWS lib